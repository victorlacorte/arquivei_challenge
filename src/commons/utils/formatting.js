function toBRL(number) {
  return number.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

export {
  toBRL,
};
