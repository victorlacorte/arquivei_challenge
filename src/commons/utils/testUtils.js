/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { ThemeProvider } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount, shallow } from 'enzyme';

import defaultTheme from 'commons/styles/theme';

function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

export {
  mountWithTheme,
  shallowWithTheme,
};
