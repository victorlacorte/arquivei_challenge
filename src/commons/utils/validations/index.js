export { default as cnpjValidations } from './cnpj';
export { default as consultKeysValidations } from './consultKeys';
export { default as creditCardValidations } from './creditCard';
export { default as formValidations } from './formValidations';
