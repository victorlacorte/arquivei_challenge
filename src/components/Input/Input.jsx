import React from 'react';
import PropTypes from 'prop-types';

import {
  StyledContainer,
  StyledInput,
  StyledP,
  StyledTitle,
} from './styles';

// Input abstraction, mainly utilized as redux-form's component prop
function Input(props) {
  const {
    input: inputProps,
    type,
    label,
    placeholder,
    meta: {
      touched,
      error,
      warning,
    },
  } = props;

  return (
    <StyledContainer>
      <StyledTitle>{label}</StyledTitle>
      <div>
        <StyledInput {...inputProps} placeholder={placeholder} type={type} error={error} />
        {touched && (
          (error && <StyledP>{error}</StyledP>) || (warning && <StyledP>{warning}</StyledP>)
        )}
      </div>
    </StyledContainer>
  );
}

Input.defaultProps = {
  input: {},

  type: 'text',

  label: '',

  placeholder: '',

  meta: {
    touched: false,
    error: '',
    warning: false,
  },
};

Input.propTypes = {
  // `input` is generic enough that we only need to ensure that is destructurable
  // eslint-disable-next-line react/forbid-prop-types
  input: PropTypes.object,

  type: PropTypes.string,

  label: PropTypes.string,

  placeholder: PropTypes.string,

  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
    warning: PropTypes.bool,
  }),
};

export default Input;
