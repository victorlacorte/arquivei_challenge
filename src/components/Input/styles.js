import styled, { css } from 'styled-components';

const StyledContainer = styled.div`
  height: 104px;
`;

const StyledInput = styled.input`
  background-color: ${({ theme }) => theme.palette.grey100.main};
  padding: 12px 32px;
  border: none;
  border-radius: 24px;
  line-height: 1.5;

  ${({ theme }) => css`
    ${theme.typography.body2};
  `}

  &:focus {
    outline: none;
  }

  ${({ error }) => error && css`
    &:focus {
      box-shadow: inset 0 0 0 2px ${({ theme }) => theme.palette.error.main};
    }
  `}
`;

const StyledTitle = styled.h2`
  ${({ theme }) => css`
    ${theme.typography.h2};
  `}

  margin-bottom: 8px;
  font-size: 18px;
`;

const StyledP = styled.p`
  ${({ theme }) => css`
    ${theme.typography.body2};
  `}

  color: ${({ theme }) => theme.palette.error.main};
  font-size: 14px;
`;

export {
  StyledContainer,
  StyledInput,
  StyledP,
  StyledTitle,
};
