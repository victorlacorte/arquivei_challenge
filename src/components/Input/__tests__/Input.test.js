import React from 'react';

import Input from 'components/Input';
import { shallowWithTheme } from 'commons/utils/testUtils';

describe('<Input />', () => {
  it('creates a Input component', () => {
    const wrapper = shallowWithTheme(<Input />);

    expect(wrapper).toMatchSnapshot();
  });
});
