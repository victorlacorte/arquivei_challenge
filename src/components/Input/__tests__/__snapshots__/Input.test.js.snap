// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> creates a Input component 1`] = `
<styles__StyledContainer>
  <styles__StyledTitle />
  <div>
    <styles__StyledInput
      error=""
      placeholder=""
      type="text"
    />
  </div>
</styles__StyledContainer>
`;
